src/logic/calculate.js

```diff
@@ -0,0 +1,14 @@
+import operate from './operate';
+
+function calculate(calculator, buttonName) {
+  let total = 0;
+  let next = 0;
+  if (buttonName === '+/-') {
+    total = calculator.total * -1;
+    next = calculator.next * -1;
+  } else if (buttonName === '+' || buttonName === '-' || buttonName === 'X' || buttonName === '%') {
+    operate(total, next, buttonName);
```

Comment:
- [ ] The operate function is called but the return value of the `operate` function is not casted into any variable. Please cast the returned value into the appropriate variable.
```
total =  operate(total, next, buttonName)
```